src/ChargeDrift/ChargeDrift.jl

```diff
@@ -118,6 +118,34 @@ function _add_fieldvector_diffusion!(step_vectors::Vector{CartesianVector{T}}, d
     end
     nothing 
 end
+function _add_fieldvector_diffusion_TiedWithMobility!(step_vectors::Vector{CartesianVector{T}}, done::Vector{Bool}, Δt::T, calculate_mobility::Function, current_pos::Vector{CartesianPoint{T}}, crystal_temperature::T, CC::Type{Electron})::Nothing where {T <: SSDFloat}
+    for n in eachindex(step_vectors)
+        if done[n] continue end
+
+        mu=calculate_mobility(current_pos[n], CC) # mobility_mu in m^2/V/s
+        D=mu*crystal_temperature*8.617e-5 # D = mobility_mu * k0*T_Deadlayer/e # D in m^2/s # k0=8.617e-5 eV/K
+        length=sqrt(6*D*Δt)
+
+        sinθ::T, cosθ::T = sincos(acos(T(2*rand() - 1)))
+        sinφ::T, cosφ::T = sincos(T(rand()*2π))
+        step_vectors[n] += CartesianVector{T}( length * cosφ * sinθ, length * sinφ * sinθ, length * cosθ )
+    end
+    nothing 
+end
+function _add_fieldvector_diffusion_TiedWithMobility!(step_vectors::Vector{CartesianVector{T}}, done::Vector{Bool}, Δt::T, calculate_mobility::Function, current_pos::Vector{CartesianPoint{T}}, crystal_temperature::T, CC::Type{Hole})::Nothing where {T <: SSDFloat}

```

Comment:
I don't see an obvious difference between the methods for `Electrons` and `Holes`, just that you pass `CC` further to `calculate_mobility`. Can you combined the two methods into one?

Reply:
Seems my understanding of the code is wrong. I thought I have to do this.
Yes, I will try.

Reply:
So, if both methods agree except for what you expect `CC` to be (but that is passed on further), so could combine both methods into one using:
```suggestion
function _add_fieldvector_diffusion_TiedWithMobility!(step_vectors::Vector{CartesianVector{T}}, done::Vector{Bool}, Δt::T, calculate_mobility::Function, current_pos::Vector{CartesianPoint{T}}, crystal_temperature::T, ::Type{CC})::Nothing where {T <: SSDFloat, CC <: ChargeCarrier}
```
see L. 249 and L.253 in https://github.com/JuliaPhysics/SolidStateDetectors.jl/blob/32851dcd3d458879dfcebd07c2352dc681dddf06/src/ChargeDrift/ChargeDrift.jl#L239-L253

Reply:
But check that your code still runs afterwards